### ISSUE TITLE
Scoped npm packages on Windows do not seem to work

### ISSUE THREAD

**Issue by Alaneor** [eslint/eslint#5644]
Hello,

it looks like scoped packages do not work correctly on Windows machines. Below you can see that eslint is trying to load a configuration from `@strv\javascript\standard\optional.js/eslint-config`, while the correct place to load it from would be `@strv\eslint-config-javascript\standard\optional.js`.

To provide more context, the file `@strv/javascript/environments/nodejs/optional.js` contains a **relative** reference to another configuration file via `extends` property:
```js
module.exports = {
  extends: '../../standard/optional.js'
}
```

If it helps, the whole configuration package is publicly available [here](https://github.com/strvcom/js-coding-standards).

I wonder if this is an issue with ESLint or with the configuration package. On OS X, this configuration package works without issues.

Thank you for your help!

**What version of ESLint are you using?** 2.4.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```js
// .eslintrc.js
module.exports = {
  extends: [
    '@strv/javascript/environments/nodejs/latest',
    '@strv/javascript/environments/nodejs/best-practices',
    '@strv/javascript/environments/nodejs/optional'
  ],

  rules: {
    'valid-jsdoc': 2
  }
}
```

**What actually happened? Please include the actual, raw output from ESLint.**

```
Cannot find module '@strv\javascript\standard\optional.js/eslint-config' from 'C:\project-root\node_modules\@strv\eslint-config-javascript\environments\nodejs'
Referenced from: @strv/javascript/environments/nodejs/optional
Referenced from: C:\project-root\.eslintrc.js
Error: Cannot find module '@strv\javascript\standard\optional.js/eslint-config' from 'C:\project-root\node_modules\@strv\eslint-config-javascript\environments\nodejs'
Referenced from: @strv/javascript/environments/nodejs/optional
Referenced from: C:\project-root\.eslintrc.js
   at Function.module.exports [as sync] (C:\project-root\node_modules\resolve\lib\sync.js:33:11)
   at resolve (C:\project-root\node_modules\eslint\lib\config\config-file.js:416:38)
   at load (C:\project-root\node_modules\eslint\lib\config\config-file.js:435:24)
   at C:\project-root\node_modules\eslint\lib\config\config-file.js:355:36
   at Array.reduceRight (native)
   at applyExtends (C:\project-root\node_modules\eslint\lib\config\config-file.js:338:28)
   at load (C:\project-root\node_modules\eslint\lib\config\config-file.js:462:22)
   at C:\project-root\node_modules\eslint\lib\config\config-file.js:355:36
   at Array.reduceRight (native)
   at applyExtends (C:\project-root\node_modules\eslint\lib\config\config-file.js:338:28)
```

/cc @MartinStava @dostalleos

**Comment by ilyavolodin** [comment]
Looks like regex for normalizing config names isn't doing the right thing

**Comment by nzakas** [comment]
Might be related to or the same as #5468.

**Comment by nzakas** [comment]
Also possibly related #5175

**Comment by ilyavolodin** [comment]
I think this is a bit different then those two issues. I'm thinking that way based on this line: ```Cannot find module '@strv\javascript\standard\optional.js/eslint-config'`` it looks like normalized name is not formatted correctly, so I'm guessing that RegEx didn't parse it properly.

**Comment by nzakas** [comment]
Working on this.

**Comment by dostalleos** [comment]
Hello,

I just updated to the latest version (2.5.3) and it's still same.

**Comment by Alaneor** [comment]
I don't think this is released yet, 2.5.3 went out before this was merged. I am assuming the fix for this issue will be available as 2.5.4 or 2.6.

**Comment by nzakas** [comment]
Yes, this fix hasn't been released yet.

**Comment by axefrog** [comment]
Please re-open this issue. It's not fixed.

```json
{
  "extends": "@frptools/config/eslint.json",
  "env": {
    "browser": true,
    "commonjs": true
  }
}
```

ERROR:

Cannot find module '@frptools/eslint-config-config/eslint.json' Referenced from: d:\Dropbox\Work\Dev\frptools\projects\salix\packages\web-client\.eslintrc.json

(symlinked package, via yarn workspaces)
